Sources/StreamChatSwiftUI/ChatThreadList/ChatThreadListViewModel.swift

```diff
@@ -0,0 +1,197 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import Foundation
+import StreamChat
+
+/// View model for the `ChatThreadListView`.
+open class ChatThreadListViewModel: ObservableObject, ChatThreadListControllerDelegate, EventsControllerDelegate {
```

Comment:
no action here, just food for thought: do you think we can reuse such view models in UIKit VCs?

Reply:
I had a plan to try this out, actually. It might be challenging, but only with a POC we will find out 🤔

Reply:
Ahah I had this in my TODO list, actually, but you made me go for it 😂